### PR TITLE
Lod group Feature

### PR DESCRIPTION
**Pull request by mycheckgithub**

### CODE REVIEW

File: COLLADAMaya/include/COLLADAMayaLODExporter.h [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+#include "COLLADAMayaPrerequisites.h"
+#include "COLLADASWStreamWriter.h"
+#include "COLLADASWLibraryNodes.h"
+#include "COLLADAMayaDocumentExporter.h"
+
+namespace COLLADAMaya
+{
+
+    /** This class writes the <library_lod>. */
```
Comment (mbarnes-sb) [inline COLLADAMaya/include/COLLADAMayaLODExporter.h]: error in comment. no such element as library_lod

File: COLLADAMaya/src/COLLADAMayaExportOptions.cpp [hunk 2]
```diff
@@ -62,8 +63,9 @@ namespace COLLADAMaya
     // Parse the options String
     void ExportOptions::set ( const MString& optionsString )
     {
+		mLOD = true;
         // Reset everything to the default value
-		mBakeTransforms = false;
+        mBakeTransforms = true;
```
Comment (mbarnes-sb) [inline COLLADAMaya/src/COLLADAMayaExportOptions.cpp]: Why is ```mBakeTransforms = true``` changing from false?

File: COLLADAMaya/include/COLLADAMayaSceneElement.h [hunk 3]
```diff
@@ -127,7 +128,7 @@ namespace COLLADAMaya
         const MObject& getNode () const;
 
         /** Returns the type of the node*/
-        const Type& getType () const;
+		const Type& getType() const;
```
Comment (mbarnes-sb) [inline COLLADAMaya/include/COLLADAMayaSceneElement.h]: gratuitous whitespace change

File: COLLADAMaya/include/COLLADAMayaSceneElement.h [hunk 4]
```diff
@@ -299,7 +300,7 @@ namespace COLLADAMaya
     private:
 
         /** Returns the type of the node*/
-        const Type determineType() const;
+        const Type determineType(/*bool cached = true*/) const;
```
Comment (mbarnes-sb) [inline COLLADAMaya/include/COLLADAMayaSceneElement.h]: non-functional change. What is ```cached``` about?

File: COLLADAMaya/include/COLLADAMayaDocumentExporter.h [hunk 5]
```diff
@@ -261,6 +275,12 @@ namespace COLLADAMaya
         */
         LightExporter* getLightExporter();
 
+		/**
+		* Returns a pointer to the light exporter.
+		* @return LightExporter* Pointer to the light exporter
```
Comment (mbarnes-sb) [inline COLLADAMaya/include/COLLADAMayaDocumentExporter.h]: these doxygen comments are wrong

File: COLLADAMaya/include/COLLADAMayaLODExporter.h [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    private:
+
+        /** Pointer to the document exporter */
+        DocumentExporter* mDocumentExporter;
+
+    public:
+
+        /**
+        * @param streamWriter The stream the output will be written to
+        * @param documentExporter The document exporter this material exporter is used in
```
Comment (mbarnes-sb) [inline COLLADAMaya/include/COLLADAMayaLODExporter.h]: these doxygen comments are wrong

File: COLLADAMaya/include/COLLADAMayaLODExporter.h [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+        */
+		LODExporter(COLLADASW::StreamWriter* streamWriter, DocumentExporter* documentExporter);
+        virtual ~LODExporter() {};
+
+        /** Exports the lights in the scene graph. */
```
Comment (mbarnes-sb) [inline COLLADAMaya/include/COLLADAMayaLODExporter.h]: these doxygen comments are wrong

File: COLLADAMaya/include/COLLADAMayaLODExporter.h (hunk 1 above)
Comment (mbarnes-sb) [inline COLLADAMaya/include/COLLADAMayaLODExporter.h]: these doxygen comments are wrong

File: COLLADAMaya/src/COLLADAMayaLODExporter.cpp [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+                mDocumentExporter->mExportPass = SECOND_LOD_PASS;
+				mVisualSceneExporter->exportVisualSceneNodes(sceneElement);
+			}
+		}
+
+        mDocumentExporter->mExportPass = VISUAL_SCENE_PASS;
+
+		closeLibrary();
+    }
+}
```
Comment (mbarnes-sb) [inline COLLADAMaya/src/COLLADAMayaLODExporter.cpp]: no newline at eof

File: COLLADAStreamWriter/src/COLLADASWNode.cpp [hunk 9]
```diff
@@ -17,9 +17,9 @@ namespace COLLADASW
 {
 
     //---------------------------------------------------------------
-    void Node::start()
+    void Node::start(bool forceNonInstancedNode)
     {
-        if ( !mIsInstanceNode )
+		if (!mIsInstanceNode || forceNonInstancedNode)
```
Comment (mbarnes-sb) [inline COLLADAStreamWriter/src/COLLADASWNode.cpp]: What is ```forceNonInstancedNode``` about?
I think this is incorrect. The proxy extra belongs to instance_node

File: COLLADAMaya/src/COLLADAMayaVisualSceneExporter.cpp [hunk 10]
```diff
@@ -642,7 +775,7 @@ namespace COLLADAMaya
         
 
         // open the scene node
-        mVisualSceneNode->start();
+		mVisualSceneNode->start(mDocumentExporter->mExportPass == SECOND_LOD_PASS);
```
Comment (mbarnes-sb) [inline COLLADAMaya/src/COLLADAMayaVisualSceneExporter.cpp]: Explain this please

File: COLLADAMaya/include/COLLADAMayaLODExporter.h (hunk 1 above)
Comment (mycheckgithub) [inline COLLADAMaya/include/COLLADAMayaLODExporter.h]: you are right, this is <library_nodes>

File: COLLADAMaya/include/COLLADAMayaSceneElement.h (hunk 4 above)
Comment (mycheckgithub) [inline COLLADAMaya/include/COLLADAMayaSceneElement.h]: sorry it's old code

File: COLLADAMaya/include/COLLADAMayaDocumentExporter.h (hunk 5 above)
Comment (mycheckgithub) [inline COLLADAMaya/include/COLLADAMayaDocumentExporter.h]: modified

File: COLLADAMaya/include/COLLADAMayaLODExporter.h (hunk 6 above)
Comment (mycheckgithub) [inline COLLADAMaya/include/COLLADAMayaLODExporter.h]: modified

File: COLLADAMaya/include/COLLADAMayaLODExporter.h (hunk 7 above)
Comment (mycheckgithub) [inline COLLADAMaya/include/COLLADAMayaLODExporter.h]: modified

File: COLLADAStreamWriter/src/COLLADASWNode.cpp (hunk 9 above)
Comment (mycheckgithub) [inline COLLADAStreamWriter/src/COLLADASWNode.cpp]: forceNonInstancedNode is a way to force to export node as an instanceNode.
With nodes in LodGroup, we want them to be exported not in the visualScene as children of lodGroupNode. We want them to be exported in libraryNode.

For example for LOD1 in the visual scene, we have : 
 <node id="_lodGroup1_pCube1_LOD1" name="pCube1_LOD1" type="NODE">

But with LOd Group we want:

<node id="LOD___lodGroup1_pCube1_LOD1" name="_lodGroup1_pCube1_LOD1" type="NODE">
      <instance_node url="#_lodGroup1_pCube1_LOD1">
 ...

so _lodGroup1_pCube1_LOD1 was Node and now is an instance Node

File: COLLADAMaya/src/COLLADAMayaVisualSceneExporter.cpp (hunk 10 above)
Comment (mycheckgithub) [inline COLLADAMaya/src/COLLADAMayaVisualSceneExporter.cpp]: This is the same as above explanation

**Pull request merged**